install-steps.md

```diff
@@ -1,193 +1,821 @@
+   ~~~sh
+   systemctl restart dnsmasq
+   ~~~
+7. Verify the `dnsmasq` service is running.
+   ~~~sh
+   systemctl status dnsmasq
+   ~~~
+
+# Install RHEL on the Provision Node
+
+With the networking portions complete, the next step in installing the 
+Red Hat OpenShift (OCP) cluster is to install RHEL 8
+on the provision node. This node will be used as the orchestrator in installing
+the OCP cluster on the 3 master and 2 worker nodes. For the purposes of this
+document, installing RHEL on the provision node is out of scope. However, options
+include, but not limited to, using a RHEL Satellite server, PXE, or installation
+media. 
+
+# Preparing the Provision node for OpenShift Install
+
+The following steps need to be performed in order to prepare the environment. 
+
+1. Login into the provision node via `ssh`
+2. Start `firewalld`, enable the `http` service, enable port 5000.
+   ~~~sh
+   systemctl start firewalld
+   firewall-cmd --zone=public --add-service=http --permanent
+   firewall-cmd --zone=public --permanent --add-service=http --permanent
+   firewall-cmd --add-port=5000/tcp --zone=libvirt  --permanent
+   firewall-cmd --add-port=5000/tcp --zone=public   --permanent
+   firewall-cmd --reload
+   ~~~
+3. Disable SELinux
+    ~~~sh
+    sed -i "s/=enforcing/=permissive/g" /etc/selinux/config
+    setenforce 0
+    ~~~
+4. Register your environment using `subscription-manager`
+   ~~~sh
+   subscription-manager register --username=<user> --password=<pass>
+   subscription-manager repos --enable=rhel-8-for-x86_64-appstream-rpms --enable=rhel-8-for-x86_64-baseos-rpms
+   ~~~
+4. Install the following packages 
+   ~~~sh
+   dnf install -y libvirt qemu-kvm mkisofs python3-devel python3-pip gcc ipmitool
+   ~~~
+5. Start and enable the `libvirtd` service
+   ~~~sh
+   systemctl start libvirtd
+   systemctl enable libvirtd --now
+   ~~~
+6. Create the `default` storage pool and start it.
+   ~~~sh
+   virsh pool-define-as --name default --type dir --target /var/lib/libvirt/images
+   virsh pool-start default
+   virsh pool-autostart default
+   ~~~
+7. Configure networking (this step can also be run from the console)
+    ~~~sh
+    # You will be disconnected, but reconnect via your ssh session after running
+    export PUB_CONN=<baremetal_nic_name>
+    export PROV_CONN=<prov_nic_name>
+    bash -x <<EOS
+    nmcli con delete "$PROV_CONN"
+    nmcli con add type bridge ifname provisioning autoconnect yes con-name provisioning stp off
+    nmcli con modify provisioning ipv4.addresses 172.22.0.1/24 ipv4.method manual
+    nmcli con add type bridge-slave autoconnect yes con-name "$PROV_CONN" ifname "$PROV_CONN" master provisioning
+    nmcli con delete "$PUB_CONN"
+    nmcli con add type bridge ifname baremetal autoconnect yes con-name baremetal stp off
+    nmcli con add type bridge-slave autoconnect yes con-name "$PUB_CONN" ifname "$PUB_CONN" master baremetal
+    systemctl restart NetworkManager
+    systemctl restart libvirtd
+    EOS
```

Comment:
So I do it this way:
    nmcli connection add ifname provisioning type bridge con-name provisioning
    nmcli con add type bridge-slave ifname "$PROV_CONN" master provisioning
    nmcli connection add ifname baremetal type bridge con-name baremetal
    nmcli con add type bridge-slave ifname "$MAIN_CONN" master baremetal
    nmcli con down "$MAIN_CONN";pkill dhclient;dhclient baremetal
    nmcli connection modify provisioning ipv4.addresses 172.22.0.1/24 ipv4.method manual
    nmcli con down provisioning
    nmcli con up provisioning
No embedded script - and by passing the pkill dclient - dhclient line all at once the connection is not lost - there will be a slight delay and then it re-establishes automatically

Reply:
I had issues with the script as is. I will investigate your way further.